tpm2/src/main/nsl_client.c

```diff
@@ -0,0 +1,181 @@
+/**
+ * @file nsl_client.c
+ * @brief A client app for testing the Needham-Schroeder-Lowe protocol
+ */
+
+#include <getopt.h>
+#include <string.h>
+#include <unistd.h>
+
+#include <kmip/kmip.h>
+
+#include "defines.h"
+#include "memory_util.h"
+#include "nsl_util.h"
+#include "socket_util.h"
+#include "kmip_util.h"
+#include "aes_gcm.h"
+
+static void usage(const char *prog)
```

Comment:
In testing this, I first tried to pass in a certificate .pem file, but had to give it a public key .pem file instead. This is OK, although I think we want to eventually use a certificate, and pass the Distinguished Name field as the ID. Therefore, although I was confused by the certificate description in the usage, I don't think we should change it. I just wanted to mention it.

Reply:
Good point. I'll clarify the usage to make sure this is clear.